processor/src/main/java/org/mapstruct/ap/internal/model/NoArgumentConstructor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.internal.model;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.mapstruct.ap.internal.model.common.ModelElement;
+import org.mapstruct.ap.internal.model.common.Type;
+
+/**
+ * Represents a constructor that is used for constructor injection.
+ *
+ * @author Sjaak Derksen
+ */
+public class NoArgumentConstructor extends ModelElement implements Constructor {
+
+    private final String name;
+    private final Set<SupportingConstructorFragment> fragments;
+
+    public NoArgumentConstructor(String name, Set<SupportingConstructorFragment> fragments) {
+        this.name = name;
+        this.fragments = fragments;
+    }
+
+    @Override
+    public Set<Type> getImportTypes() {
+        return Collections.emptySet();
```

Comment:
Thought about that. You could claim the same for the instance variable definition. In the end I considered both fragment as instance variable definition as supportive to the built-in mapping method that uses them. So that defines all. It's perhaps not entirely perfect from a separation of concerns perspective but it'll do 😀